### DISCUSSION TITLE
RFC: Dapp Treasury Usage

### DISCUSSION THREAD

**Discussion by zanicar** [archway-network/discussions#22] [Ideas]
# RFC: Dapp Treasury Usage

![Treasury Banner](https://github.com/archway-network/discussions/blob/main/media/jouwen-wang-UChknR2z5EM-unsplash-cropped.jpg?raw=true)
Photo by <a href="https://unsplash.com/@sodacheese?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText">Jouwen Wang</a> on <a href="https://unsplash.com/photos/UChknR2z5EM?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText">Unsplash</a>

## Abstract

We are re-evaluating the usage of the Dapp Treasury and would appreciate feedback from both the general Archway community and dapp developer community in this regard.

We have identified the following three paths to manage the treasury’s future token distribution and would like to ascertain which resonates with our community before making definitive design decisions:

1. Algorithmic
2. General Governance
3. Special Interest Group (Mini DAO)

This RFC will remain open to the end of September 2023.

## Background

The Dapp Treasury was originally envisioned as a sidecar mechanism to Developer Inflationary Tokens (DIT). _DIT represents the portion of inflationary rewards allocated to dapps_, but dapps earn DIT proportional to their individual block fill rate, with certain limitations. Thus, not all DIT are allocated to specific dapps for every block, and _the Dapp Treasury acts as a collector of the remaining unallocated tokens_. These tokens are still earmarked for the benefit of dapps and dapp developers.

Archway is actively pursuing the creation and stimulation of a sustainable smart contract network, which means we not only look at what is currently possible, but also what may be possible in the near and not so near future. Consequently, our vision and designs reflect this philosophy... Just because something is not possible or feasible today does not mean it will be discarded or ignored. In this vein our economic designs contain certain features and mechanisms that cannot currently be expressed to their full potential, but we do work to match it as closely as possible. However, as the ecosystem matures and technology evolves we work toward the realization of these ideals.

- You can learn more about our designs [here](https://archway.io/whitepapers).
- You can also review our current technical specifications [here](https://github.com/archway-network/archway/tree/main/docs).
- You can join the discussion on the current state of inflationary rewards [here](https://github.com/orgs/archway-network/discussions/10).

### Algorithmic

Algorithmic distribution entails allocation of dapp treasury tokens to dapps based on some mathematical formula to rank or select the "best performing" dapps on a periodic basis. "Best performing" may be anything from largest average transaction volume to greatest increase in transaction volume over the determined period.

**Advantages:** Fair, simple and deterministic process; The performance metric can be selected to avoid potential gaming of the mechanism;

**Disadvantages:** May favor a specific subset of dapps based on the performance metric; Lacks flexibility to be quickly adjusted to environmental changes or stimuli;

### General Governance

General governance-based distribution entails the utilization of the general on-chain governance mechanism to determine via community-driven custom proposals how to allocate dapp treasury tokens.

**Advantages:** Utilizes a familiar governance framework within clearly defined participation parameters and expectations;

**Disadvantages:** May result in a Moral Hazard situation where the objectives of dapp developers and general network users are opposed, leaving dapp developers at a distinct disadvantage; Slow, cumbersome process;

### Special Interest Group (Mini DAO)

Mini DAO based distribution entails the formation of a special interest group represented by an on-chain DAO to manage the token distribution of the dapp treasury.

**Advantages:** DAO may utilize an election / selection mechanism to counteract the potential for Moral Hazard situations by ensuring it represents relevant stakeholders across the ecosystem; Small task-focused group with greater speed and flexibility;

**Disadvantages:** Smaller intimate group may be more susceptible to collusion and antitrust behavior; Lack of diverse ecosystem representation may quickly result in favoritism;

## Discussion

We would like to know the community's take on the above. Please feel free to raise additional options regarding distribution. However, we ask that you are considerate of the potential technical implications, such as complexity, cost and time scales, so do include some analysis on these metrics if possible.

**Comment by jlm0** [comment]
The ideal scenario would be an algorithmic distribution mechanism that is adjustable in real-time by a DAO composed of diverse ecosystem stakeholders. This combines fairness through metrics-based allocation with flexibility through community oversight.

However, given current technical constraints, a pragmatic approach could be establishing a DAO committee focused on managing the treasury distribution. This DAO should still leverage clear metrics to guide allocation decisions, avoiding pure subjective judgment. The selection process for DAO members is critical - rules and procedures need to be established upfront to ensure no single interest group dominates.

As DAO adoption increases across networks, following this model of specialized DAOs handling targeted governance makes sense. However, care should be taken to avoid duplicating the pitfalls of legacy governance systems. Procedural transparency and regular rotation of DAO members will be important to maintain credibility.

Overall I think a data-driven DAO with diverse representation, following clear operational processes, provides a balanced mechanism for managing the dapp treasury until an adjustable algorithmic approach becomes technically feasible. The DAO should be empowered but also held accountable through its design and governance.

Thanks for the discussion @zanicar. I'll duplicate onto the governance forum as I know Eric was looking for feedback on the same topic!
